crates/aptos/src/common/types.rs

```diff
@@ -298,6 +301,16 @@ impl KeyInputOptions {
             }
         }
     }
+
+    pub fn extract_x25519_public_key(
```

Comment:
There might be some rust magic way to combine `extract_public_key` and `extract_x25519_public_key` in a nice way with generics?

Reply:
There only would be if there was a `From<x25519::PublicKey> for Ed25519PublicKey` implemented.  Then, you could just put an `into()` in the middle with some generics, and it should just copy it.  But, I think that's out of the scope of this for now, and not needed, mostly because it needs more error handling.